### DISCUSSION TITLE
How can use time range of grafana in UQL or Parsing options?

### DISCUSSION THREAD

**Discussion by seasoncool** [grafana/grafana-infinity-datasource#543] [Q&A]
I want to filter the time by grafana time range in UQL or Parsing, I can't make it happen, please help me out. thanks in advance.

UQL:  I want to filter last 30 days. 
`parse-json
| scope "value"
| project "time"="fields.['System.CreatedDate']"
`
![image](https://user-images.githubusercontent.com/6467295/222743278-7996991b-962b-44a2-aaaf-549b21d68ba9.png)

same request for 30 days with parsing.

![image](https://user-images.githubusercontent.com/6467295/222743914-d61989e8-8b81-4cb9-8848-90042799e67d.png)

**Comment by yesoreyeram** [comment]
you have to pass the time range parameters to the url. https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#__from-and-__to

**Reply by seasoncool** [reply]
@yesoreyeram thanks for your quick response.

how about UQL, can i make it happened with UQL ?

time range parameters to the url -> my json file is static single file on server(like: http://10.126.213.59/points.json), unable to pass parameters :(

could you give me a detailed example？ thanks.

**Reply by chicago-joe** [reply]
I have the same situation as above ^. If you have any suggestions, please let me know!
